Mathlib/Data/Nat/Cast/Basic.lean

```diff
@@ -183,6 +183,9 @@ variable {π : α → Type*} [∀ a, NatCast (π a)]
 
 instance instNatCast : NatCast (∀ a, π a) where natCast n _ := n
 
+instance instOfNat (n : ℕ) [∀ i, OfNat (π i) n] : OfNat ((i : α) → π i) n where
+  ofNat _ := OfNat.ofNat n
```

Comment:
Can you move this and the lemmas about it to a new section? This section assumes `[∀ a, NatCast (π a)]`, which you don't want here.